rescript/tests/test_get_unite.py

```diff
@@ -0,0 +1,87 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2019-2023, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import qiime2
+import pkg_resources
+import tempfile
+from qiime2.plugin.testing import TestPluginBase
+from qiime2.plugins import rescript
+from rescript.get_unite import (
+    _unite_get_doi,
+    _unite_get_url,
+    _unite_get_tgz,
+    _unite_get_artifacts,
+)
+
+from urllib.request import urlopen
+from urllib.error import HTTPError
+from unittest.mock import patch
+
+
+class TestGetUNITE(TestPluginBase):
+    package = "rescript.tests"
+
+    def test_unite_get_doi(self):
+        # All supported inputs to _unite_get_doi
+        versions = ["9.0", "8.3", "8.2"]
+        taxon_groups = ["fungi", "eukaryotes"]
+        singletons = [True, False]
+        # Eventual output
+        obs_dois = []
+        # for all combinations
+        for v in versions:
+            for tg in taxon_groups:
+                for s in singletons:
+                    # add it to the list
+                    obs_dois.append(_unite_get_doi(v, tg, s))
+        exp_dois = [
```

Comment:
I don't like hard-coding the list of valid DOIs in two places.

If a dict lookup fails, it will throw an error. Maybe simply looking up all combinations is enough?

This function is also used within `_unite_get_url`, maybe we just include it there?

Reply:
okay...

I've split out this into a global variable. This test still seems circular/trivial to me.

If dict lookup is trivial, maybe we should only have `_unite_get_url` and the real test if the URLs can be found. Like Nick mentioned, we constrain input variables in the CLI.

Reply:
I have removed `_get_unite_doi()` and added DOI lookup as an internal function `_unite_get_url`